azext_iot/central/models/devicetwin.py

```diff
@@ -0,0 +1,110 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import re
+from datetime import datetime, timezone, timezone
+from azext_iot.central.models.template import Template
+
+
+class DeviceTwin:
+    def __init__(
+        self, device_twin: dict,
+    ):
+        self.device_twin = device_twin
+        self.device_id = device_twin.get("deviceId")
+        self.desired_property = Property(
+            "desired property",
+            device_twin.get("properties").get("desired"),
```

Comment:
if the aim is safety, you have to provide a default value:

```
device_twin.get("properties", {}).get("desired", {})
```

same change needed elsewhere